File: scripts/r/interpolacion_lineal_series_con_datos_faltantes.R
df <- df

df$variable_incompleta[df$variable_incompleta == 'X'] <- NA
df$variable_incompleta <- as.numeric(df$variable_incompleta)
df$variable_incompleta <- approx(x = df$anio, y = df$variable_incompleta, xout = df$anio)$y

print(df)

File: scripts/python/interpolacion_lineal_series_con_datos_faltantes.py
import pandas as pd
import numpy as np

df = df

df["variable_incompleta"] = df["variable_incompleta"].replace("X", np.nan).astype(float)
df["variable_incompleta"] = df.set_index("anio")["variable_incompleta"].interpolate(method="index").values

print(df)
